packages/tldraw/src/lib/utils/export/exportAs.ts

```diff
@@ -18,15 +19,19 @@ export async function exportAs(
 	editor: Editor,
 	ids: TLShapeId[],
 	format: TLExportType = 'png',
+	name: string | undefined,
 	opts = {} as Partial<TLSvgOptions>
```

Comment:
(general observation, not feedback on this PR) if we standardised on the single-object style function args we would be able to make changes like this without breakage. I wish I had pushed for that more often.

Reply:
Clojure all the way 😄 

It's also confusing that we have opts, but then on top of that we also have a bunch of args 🤷

Reply:
it's not too late. There is a migration path for functions/methods with multiple args to a single arg, using overloads. something like this (rough sketch)

```ts
function myFunction(the: number, old: number, params: number): Blah
function myFunction(args: {the: number, old: number, params: number, andTheNewOne?: number}): Blah
function myFunction() {
  let the, old, params, andTheNewOne
  if (arguments.length === 1) {
     ;({the, old, params, andTheNewOne}) = arguments[0]
  } else {
     the = arguments[0]
     old = arguments[1]
     params = arguments[2]
     andTheNewOne = undefined
  }

  ....
}
```

Reply:
at the same time you could add a console.warn for the old signature to make people switch over, and then eventually delete the old version.